insecure/integration/functional/test/gossipsub/rpc_inspector/validation_inspector_test.go

```diff
@@ -894,6 +943,109 @@ func TestValidationInspector_InspectRpcPublishMessages(t *testing.T) {
 	require.Equal(t, uint64(1), notificationCount.Load())
 }
 
+// TestValidationInspector_MultiErrorNotification ensures that control message validation captures all errors for invalid control messages for a single control message type. RPC validation is performed
+// on each control message type, during this validation a sample of the control messages for that type are inspected. This test ensures that the entire sample is inspected and all errors are captured in
+// the p2p.InvCtrlMsgErrs field of the invalid control message notification.
+func TestValidationInspector_MultiErrorNotification(t *testing.T) {
```

Comment:
The test scenario you've outlined is comprehensive, yet it would be extremely beneficial to expand it to encompass every possible combination of control messages and error types for a more thorough evaluation.

Reply:
https://github.com/onflow/flow-go/pull/4978/commits/aa6a096c39c987ce1f1c97bf69422a82589a1f0a